lib/msf/core/payload/windows/reverse_http.rb

```diff
@@ -0,0 +1,339 @@
+# -*- coding: binary -*-
+
+require 'msf/core'
+require 'msf/core/payload/windows/block_api'
+require 'msf/core/payload/windows/exitfunk'
+
+module Msf
+
+
+###
+#
+# Complex payload generation for Windows ARCH_X86 that speak HTTP(S)
+#
+###
+
+
+module Payload::Windows::ReverseHttp
+
+  include Msf::Payload::Windows::BlockApi
+  include Msf::Payload::Windows::Exitfunk
+
+  #
+  # Register reverse_http specific options
+  #
+  def initialize(*args)
+    super
+    register_advanced_options(
+      [
+        OptInt.new('HTTPStagerURILength', [false, 'The URI length for the stager (5 to 240ish bytes)'])
```

Comment:
Since the URL shares space with the host (or hostname), port, and URL, it has to be more of an estimate than anything else. The runtime checks for final URL length handle cases where a long URI with a long hostname don't fit.